test/shelf_io_test.dart

```diff
@@ -42,21 +42,6 @@ void main() {
     expect(response.body, 'Hello from /');
   });
 
-  test('sync null response leads to a 500', () async {
-    await _scheduleServer((request) => Response.internalServerError());
-
-    var response = await _get();
-    expect(response.statusCode, HttpStatus.internalServerError);
-    expect(response.body, 'Internal Server Error');
-  });
-
-  test('async null response leads to a 500', () async {
```

Comment:
isn't this a valid test in sound mode?

Reply:
It passes for the wrong reason.

What this test demonstrates in sound mode is that the `Future.value` constructor has a bad API for null safety, and the `Future.value(null)` is acting like a `Future.error` in sound mode. We already have that behavior covered in sound mode by the test `'async error leads to a 500'`.

Reply:
👍

Reply:
I'll use `async => null` instead of `Future.value(null)` to make this more clear why it's only an opt-out test. The former would have a static error in opt-in code.